### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by cooljbg** [JKorf/Bittrex.Net#15]
Would you add subscribe to all markets instead subscribing to them in foreach?
Something like:
bool subscribeToall = false;
private Action<BittrexMarketSummary> onUpdateAll;

public async Task<BittrexApiResult<int>> SubscribeToAllMarketDeltaStream(Action<BittrexMarketSummary> onUpdate){
  subscribeToAllSummaries = true;
 onUpdateAll = onUpdate;
...........
}

private void SocketMessage(IList<JToken> jsonData){
if (subscribeToAllSummaries) {
                    foreach (var delta in data.Deltas)
                        onUpdateAll(delta);
                }
                else
                {
                    foreach (var delta in data.Deltas)
                        foreach (var update in registrations.Where(r => r.MarketName == delta.MarketName))
                            update.Callback(delta);
                }
}

**Comment by JKorf** [comment]
I'll look into this, probably will be useful yes.

**Comment by cooljbg** [comment]
Also subscribe for OrderBook:
- Initial
- Updates

When you change the market, you have to clean the previous market orders subscibtion, but since Bittrex doesn't have unsubscribe command, we will need a new proxy instance and cleanup for the previous one. ( the proxy instance for "updateSummaryState" will continue to work on market change because it's different instance)
Disposition of proxy or connection takes 30 seconds - its a timeout bug or feature of SignalR client. So   proxyOrders = null is better than proxyOrders.Dispose() and relying on GarbageCollector to clean it.

Example code:
private static IHubProxy proxyOrders;
private static IDisposable ordersSubscribtion; 
private const string UpdateOrdersEvent = "updateExchangeState";
private const string InitialOrdersCmd = "QueryExchangeState";
private const string UpdateOrdersCmd = "SubscribeToExchangeDeltas";


 public async Task<bool> SubscribeToMarketOrdersStream(string marketName, Action<dynamic> onInit, Action<dynamic> onUpdate){

 if (proxyOrders != null){
  ordersSubscribtion.Dispose();
  proxyOrders = null;  
}
  proxyOrders = connection.CreateHubProxy(HubName);
  ordersSubscribtion = proxy.On(UpdateOrdersEvent , onUpdate);
  await proxyOrders.Invoke(UpdateOrdersCmd, marketName);
  var existingOrders = proxy.Invoke<dynamic>(InitialOrdersCmd, marketName);
  onInit(existingOrders.Result);
}

**Comment by JKorf** [comment]
I've looked into the order book subscription, but found some problems. It isn't possible to create a new proxy when a connection has already been made. This means we can't cleanup the previous subscription when creating a new one, or when it isn't needed anymore, if we also subscribed to the market delta stream. This means that once you've subscribed to the orderbook the connection will keep sending the orderbook updates until the connection gets closed.

If you find a way around this I'll be glad to implement it, but for now I'll leave it be.
